dist/io/processTemplate.js

```diff
@@ -1,12 +1,17 @@
 "use strict";
 Object.defineProperty(exports, "__esModule", { value: true });
-function processTemplate(content, variables) {
-    let node = content;
-    for (const variable in variables) {
-        node = node.split(`{{${variable}}}`).join(String(variables[variable]));
-        node = node.split(`{{ ${variable} }}`).join(String(variables[variable]));
-    }
-    return node;
+function processTemplate(content, variables = {}) {
+    // Inject the variables in a the global __VARIABLES__
+    const injectedVariables = `<script type="text/javascript">window.__VARIABLES__ = ${JSON.stringify(variables)}</script>`;
+    // Replace all occurences of {{ string }} with the variable (empty if not found)
+    const replacedOccurrences = content.replaceAll(/({{)([0-9a-zA-Z-\s]+)(}})/g, (fullMatch, match1, match2) => {
+        const variable = match2.trim();
+        const value = variable in variables ? variables[variable] : '';
+        if (typeof value === "object")
+            return JSON.stringify(value);
+        return String(value);
+    });
```

Comment:
Since `JSON.stringify` can throw exception, we should probably wrap this in a try/catch and log so the dev knows we failed to parse their variables, what do you think?

Reply:
Good idea !